src/v/datalake/record_multiplexer.h

```diff
@@ -55,11 +55,36 @@ class record_multiplexer {
       table_creator&,
       model::iceberg_invalid_record_action,
       location_provider,
-      translation_probe&,
-      lazy_abort_source& as);
+      translation_probe&);
 
-    ss::future<ss::stop_iteration> operator()(model::record_batch batch);
-    ss::future<result<write_result, writer_error>> end_of_stream();
+    /**
```

Comment:
nit: maybe worth bringing part of the commit message into a comment in this class too:

```
mux = create_mux();
mux.multiplex(reader1...)
mux.flush_writers(); // optional
mux.multiplex(reader2..)
mux.flush_writers(); // optional
...
...

result = co_await std::move(mux).finish();
```

At least, I found it really helpful in understanding the expected new code structure and I imagine it would be useful for other new readers of the code

Reply:
done